libc/src/string/memory_utils/utils.h

```diff
@@ -293,7 +293,7 @@ LIBC_INLINE void store64_aligned(uint64_t value, Ptr dst, size_t offset) {
 // Advances the pointers p1 and p2 by offset bytes and decrease count by the
 // same amount.
 template <typename T1, typename T2>
-LIBC_INLINE void adjust(ptrdiff_t offset, T1 *__restrict &p1,
+LIBC_INLINE void adjust(uintptr_t offset, T1 *__restrict &p1,
```

Comment:
Hello - there's a call to `adjust` that passes a negative value here: https://github.com/llvm/llvm-project/blob/main/libc/src/string/memory_utils/op_generic.h#L270.  With this code change, this can raise an overflow error on asan builds. Could you take a second look at the callers of this function and make adjustments?

Reply:
Indeed, AFAIK `ptrdiff_t` is the only correct type here.

Reply:
I see.

I (incorrectly) concluded that adding to count was malformed, probably due to mixing of ptrdiff_t and size_t

Strange that the warnings didn't catch the negative argument

Will fix it